### PR TITLE
[Release] v0.9.0 release note draft.

### PR DESCRIPTION
**Pull request by wangcan0329**
I draft v0.9 release note and each service owner please kindly help update the note.

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Release v0.9.0
+
+## New Features
```
Comment (fanyangCS) [inline release-notes.md]: >Features [](start = 7, length = 8)

* new REST APIs to update Virtual Cluster
* Support training on RDMA network for on-premise clusters

File: release-notes.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* Add pai service dashboard to grafana. - [PR 1694](https://github.com/Microsoft/pai/pull/1694)
+* Webportal plugin implementation and change pai marketplace to plugin. -[PR 1700](https://github.com/Microsoft/pai/pull/1700)
+* Support update virtual cluster dynamically from webportal. -[PR 1831](https://github.com/Microsoft/pai/pull/1831) -[PR 1974](https://github.com/Microsoft/pai/pull/1974)
+
+
+## Improvements
+### Service
```
Comment (fanyangCS) [inline release-notes.md]: support hdfs path customization?

File: release-notes.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+* Cluster object model implementation to make it easier for developer to add customized service configuration generation logic. -[PR 1735](https://github.com/Microsoft/pai/pull/1735)
+* Job exporter refactor to avoid single external command call will make exporter hang indefinitely. -[PR 1840](https://github.com/Microsoft/pai/pull/1840)
+* Make drivers version configurable. - [PR 1626](https://github.com/Microsoft/pai/pull/1626)
```
Comment (fanyangCS) [inline release-notes.md]: >drivers [](start = 7, length = 7)

GPU driver

File: release-notes.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## New Features
+* Add pai service dashboard to grafana. - [PR 1694](https://github.com/Microsoft/pai/pull/1694)
+* Webportal plugin implementation and change pai marketplace to plugin. -[PR 1700](https://github.com/Microsoft/pai/pull/1700)
+* Support update virtual cluster dynamically from webportal. -[PR 1831](https://github.com/Microsoft/pai/pull/1831) -[PR 1974](https://github.com/Microsoft/pai/pull/1974)
+
+
+## Improvements
+### Service
+* Cluster object model implementation to make it easier for developer to add customized service configuration generation logic. -[PR 1735](https://github.com/Microsoft/pai/pull/1735)
+* Job exporter refactor to avoid single external command call will make exporter hang indefinitely. -[PR 1840](https://github.com/Microsoft/pai/pull/1840)
```
Comment (xudifsd) [inline release-notes.md]: User can skip driver installation if they had it pre-installed (#1841)

File: release-notes.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## New Features
+* Add pai service dashboard to grafana. - [PR 1694](https://github.com/Microsoft/pai/pull/1694)
+* Webportal plugin implementation and change pai marketplace to plugin. -[PR 1700](https://github.com/Microsoft/pai/pull/1700)
+* Support update virtual cluster dynamically from webportal. -[PR 1831](https://github.com/Microsoft/pai/pull/1831) -[PR 1974](https://github.com/Microsoft/pai/pull/1974)
+
+
+## Improvements
+### Service
+* Cluster object model implementation to make it easier for developer to add customized service configuration generation logic. -[PR 1735](https://github.com/Microsoft/pai/pull/1735)
```
Comment (mzmssg) [inline release-notes.md]: Move user code to a background process
https://github.com/Microsoft/pai/pull/1461

File: release-notes.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+* Job exporter refactor to avoid single external command call will make exporter hang indefinitely. -[PR 1840](https://github.com/Microsoft/pai/pull/1840)
+* Make drivers version configurable. - [PR 1626](https://github.com/Microsoft/pai/pull/1626)
+* Extend yarn local log expiration time to 7 days. -[PR 1673](https://github.com/Microsoft/pai/pull/1673)
+* Add two driver images. Current supports driver's versions are 384.111, 390.25 and 410.73. By default will deploy 390.25 version. -[PR 1642](https://github.com/Microsoft/pai/pull/1642)
+* Reduce grafana image from 440M to 280M by merging all startup scripts add removing useless plugin. -[PR 1685](https://github.com/Microsoft/pai/pull/1685)
+* Upgrade Nodejs version of webportal and rest server to 8. -[PR 1453](https://github.com/Microsoft/pai/pull/1453)
+
```
Comment (ydye) [inline release-notes.md]: Configuration Storage https://github.com/Microsoft/pai/commit/caee4a56d3674c5b5653662f7cf3cdb57fa3e055 https://github.com/Microsoft/pai/commit/2ec8f9a6b30355a1775aeb1fef7639331a88100d https://github.com/Microsoft/pai/commit/a47ed4aac62340d7ccb78557935772613bf05db3
https://github.com/Microsoft/pai/commit/fcb41fc9fe85124dc82c27ae18f42791f3b6fc32

File: release-notes.md (hunk 1 above)
Comment (fanyangCS) [inline release-notes.md]: support customer environment variables?

File: release-notes.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Release v0.9.0
+
+## New Features
+* Add pai service dashboard to grafana. - [PR 1694](https://github.com/Microsoft/pai/pull/1694)
+* Webportal plugin implementation and change pai marketplace to plugin. -[PR 1700](https://github.com/Microsoft/pai/pull/1700)
```
Comment (Gerhut) [inline release-notes.md]: Plugins doc: https://github.com/Microsoft/pai/blob/master/docs/webportal/PLUGINS.md

File: release-notes.md (hunk 1 above)
Comment (hao1939) [reply release-notes.md]: @wangcan0329 , please add "support job env variables', PR #1544

File: release-notes.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+* Make drivers version configurable. - [PR 1626](https://github.com/Microsoft/pai/pull/1626)
+* Extend yarn local log expiration time to 7 days. -[PR 1673](https://github.com/Microsoft/pai/pull/1673)
+* Add two driver images. Current supports driver's versions are 384.111, 390.25 and 410.73. By default will deploy 390.25 version. -[PR 1642](https://github.com/Microsoft/pai/pull/1642)
+* Reduce grafana image from 440M to 280M by merging all startup scripts add removing useless plugin. -[PR 1685](https://github.com/Microsoft/pai/pull/1685)
+* Upgrade Nodejs version of webportal and rest server to 8. -[PR 1453](https://github.com/Microsoft/pai/pull/1453)
+
+### Job
+* Add timestamp for cloned job's name - [PR 1532](https://github.com/Microsoft/pai/pull/1532)
+* Add log if job's image doesn't have ssh server - [PR 1675](https://github.com/Microsoft/pai/pull/1675)
+
```
Comment (hao1939) [inline release-notes.md]: We add an example demonstrating howto integrate jupyter and PAI using restserver.

> The example is a prototype of online job editor. It demonstrates:
> 
>  1. Launcher a jupyter notebook for online code editing and interactive debugging. The editor is running on PAI, so user will debug on the same environment as the job will be running and get the feedback on the fly.
> 2. Leverage PAI restserver APIs to submit job on PAI. So user could try more experiments without leaving the editor.

Comment (Gerhut) [inline release-notes.md]: https://github.com/Microsoft/pai/pull/1860

Escape injected variables in shell scripts

File: release-notes.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
+# Release v0.9.0
+
+## New Features
+* Add pai service dashboard to grafana. - [PR 1694](https://github.com/Microsoft/pai/pull/1694)
```
Comment (scarlett2018) [inline release-notes.md]: @xudifsd  may we have an introduction for the PAI Service Dashboard, there is only one picture in PR 1694 now.

File: release-notes.md (hunk 7 above)
Comment (scarlett2018) [reply release-notes.md]: Thanks @Gerhut , this is great help. Suggest to change the ReleaseNote as below:

Support to add custom web pages to the web portal of PAI deployments with **WebPortal Plugin**, refer to [Plugins Doc]( https://github.com/Microsoft/pai/blob/master/docs/webportal/PLUGINS.md) for how to use the new feature, and refer to [PR 1700](https://github.com/Microsoft/pai/pull/1700) for how PAI Marketplace is using it as an example.

File: release-notes.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+* Webportal plugin implementation and change pai marketplace to plugin. -[PR 1700](https://github.com/Microsoft/pai/pull/1700)
+* Support update virtual cluster dynamically from webportal. -[PR 1831](https://github.com/Microsoft/pai/pull/1831) -[PR 1974](https://github.com/Microsoft/pai/pull/1974)
+
```
Comment (scarlett2018) [inline release-notes.md]: Can anyone add the doc intro for this feature? thanks.

File: release-notes.md (hunk 9 above)
Comment (xudifsd) [reply release-notes.md]: Maybe "Cluster admin can now view resource consumption by pai services in PaiServiceMetrics page"?

File: release-notes.md (hunk 2 above)
Comment (wangcan0329) [reply release-notes.md]: Added

**Comment by mzmssg** [comment]
Should add a note that user need to add vc manually after upgrading

File: release-notes.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+* Add config sub-command in paictl to manage config files. -[PR 1263](https://github.com/Microsoft/pai/pull/1263)
+* After the configuration storage is enabled -p is no longer required in the service sub-command.
+
+
+## Bug Fixes
+* [issue 1603](https://github.com/Microsoft/pai/issues/1603) is fixed by adding job_exporter_iteration_seconds to expose iteration time.  - [PR 1627](https://github.com/Microsoft/pai/pull/1627).
+* [issue 1602](https://github.com/Microsoft/pai/issues/1602) is fixed by initializing the host ip from None to unscheduled - [PR 1625](https://github.com/Microsoft/pai/pull/1625).
+* [issue 1639](https://github.com/Microsoft/pai/issues/1639) is fixed by adding imagePullSecrets to prometheus. - [PR 1678](https://github.com/Microsoft/pai/pull/1678).
+* [issue 1600](https://github.com/Microsoft/pai/issues/1600) is fixed by offloading docker daemon check from watchdog to job-exporter. - [PR 1670](https://github.com/Microsoft/pai/pull/1670).
+* Fix admin can't submit job to newly added virtaul cluster. - [PR 1972](https://github.com/Microsoft/pai/pull/1972)
```
[the remark this answers is not in the document]
Comment (xudifsd) [reply release-notes.md]: @wangcan0329 have not added this?

**Comment by fanyangCS** [comment]
please acknowledge external contribution.

**Pull request closed without merging**